packages/zosfiles/src/methods/copy/Copy.ts

```diff
@@ -84,10 +85,19 @@ export class Copy {
 
         const targetDataSetExists = await this.dataSetExists(session, toDataSetName);
 
-        const newDataSet = !targetDataSetExists;
+        let newDataSet = !targetDataSetExists;
+
+        const overwriteDataset = options.overwrite;
+
+        if(overwriteDataset){
+            await Delete.dataSet(session,toDataSetName);
+            newDataSet = true;
+        }
```

Comment:
`newDataSet` can be made const:

```suggestion
        const overwriteDataset = options.overwrite;

        if(overwriteDataset){
            await Delete.dataSet(session,toDataSetName);
        }
        const newDataSet = !targetDataSetExists || overwriteDataset;
```